R/decisiontreeutils.R

```diff
@@ -82,30 +81,67 @@ createDTParams <- function(config, data) {
     surrogate <- 2
   }
 
-  rpart_params$usesurrogate <- rxDTree_params$useSurrogate <- surrogate
+  param_list$usesurrogate <- surrogate
 
   # get max bins param
   if(is_XDF && !is.na(as.numeric(config$max.bins))) {
     max_bins <- config$max.bins
     if(max_bins < 2) {
       stop("The minimum bins is 2")
     } else {
-      rxDTree_params$maxNumBins <- max_bins
+      param_list$maxNumBins <- max_bins
     }
   }
 
   # other parameters
-  rpart_params$minsplit <- rxDTree_params$minSplit <- config$min.split
-  rpart_params$minbucket <- rxDTree_params$minBucket <- config$min.bucket
-  rpart_params$xval <- rxDTree_params$xVal <- config$xval.folds
-  rpart_params$maxdepth <- rxDTree_params$maxDepth <- config$max.depth
+  param_list$minsplit <- config$min.split
+  param_list$minbucket <- config$min.bucket
+  param_list$xval <- config$xval.folds
+  param_list$maxdepth <- config$max.depth
 
-  rpart_params$cp <- rxDTree_params$cp <- ifelse(config$the.cp == "Auto" || config$the.cp == "", .00001, config$the.cp)
+  param_list$cp <- ifelse(config$the.cp == "Auto" || config$the.cp == "", .00001, config$the.cp)
 
-  f <- ifelse(is_XDF, "rxDTree", "rpart")
-  args <- ifelse(is_XDF, rxDTree_params, rpart_params)
+  param_list
+}
 
-  list(f = f, args = args, is_XDF = is_XDF)
+#' name mapping from parameters to functions
+#'
+#' @param f_string string of function
+#' @param param_list list of decision tree params
+#' @return list with named parameters for f_string
+paramsToDTArgs <- function(f_string, param_list) {
+  if (f_string == "rpart") {
+    list(
+      data = param_list$data,
+      formula = param_list$f,
+      weights = param_list$weight_arg,
+      method = param_list$method,
+      parms = param_list$parms,
+      usesurrogate = param_list$surrogate,
+      minsplit <- param_list$minsplit,
+      minbucket = param_list$minbucket,
+      xval = param_list$xval,
+      maxdepth = param_list$maxdepth,
+      cp = param_list$cp
+    )
```

Comment:
I have a suggestion. Why not keep the names in `param_list` consistent with what is expected by R. There is no advantage to be gained (as far as I can see) by naming things differently (e.g. `weight_arg` instead of `weights` or `usesurrogate` instead of `surrogate`). This will reduce the number of variable names and make the code easier to grep IMHO. You can eliminate lines 114 - 125 if you did that.